src/WebJobs.Script.WebHost/Metrics/FlexConsumptionMetricsPublisher.cs

```diff
@@ -90,11 +95,13 @@ internal async Task OnPublishMetrics()
         {
             try
             {
-                if (FunctionExecutionCount == 0 && FunctionExecutionTimeMS == 0 && !IsAlwaysReady)
-                {
-                    // no activity to report
-                    return;
-                }
+                // This is publishing metrics only when there are function executions, this would not work
```

Comment:
Consider having a method `_metricsCollector.HasMetrics()` that could be included in this check to avoid writing metrics when not necessary